onadata/apps/api/tests/viewsets/test_abstract_viewset.py

```diff
@@ -244,6 +251,13 @@ def _publish_xls_form_to_project(self, publish_data={}, merge=True,
                 post_data = {'xls_file': xls_file}
                 request = self.factory.post(
                     '/', data=post_data, **self.extra)
+                request.META['HTTP_DATE'] = '2020-09-10T11:56:32.424726+00:00'
+                request.META['HTTP_REFERER'] = settings.HOSTNAME +\
+                    f':8000'
+                request.META['HTTP_USER_AGENT'] =\
+                    'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit'\
+                    '/537.36 (KHTML, like Gecko) Chrome'\
+                    '/83.0.4103.61 Safari/537.36'
```

Comment:
Why was this necessary to add?

Reply:
Since we introduced tracking newly created forms and projects, a KeyError is raised when making this request on the test.
I'd figure real time requests would have this already set, so I updated the test to include these header variables

Reply:
Mihm instead of adding this in, could we change [these lines](https://github.com/onaio/onadata/pull/1895/files#diff-9ddcf8387d802e0eed517864f979f2ff08dd99dbfabf287382548c1ef5771bfeR192-R197) to use `.get()`? In some cases, the requests may not have those values I believe...